Explain where DotNetPublishFiles and ResolvedFileToPublish come from (#12946)
aspnet/AspNetCore.Docs: aspnetcore/host-and-deploy/visual-studio-publish-profiles.md

## Include files

> ### Simple file inclusion 

Author: I'm open to suggestions for this heading.

> ### Simple file inclusion 

Reviewer: "*Simple*" ... Isn't that on the *banned word list*. :smile: lol

```suggestion
### General file inclusion
```

> The following example's `<ItemGroup>` element demonstrates copying a folder located outside of the project directory to a folder of the publish site. Any files added to the following markup's `<ItemGroup>` are included by default.

Reviewer: ```suggestion
The following example's `<ItemGroup>` element demonstrates copying a folder located outside of the project directory to a folder of the published site. Any files added to the following markup's `<ItemGroup>` are included by default.
```

```xml
<ItemGroup>
  <_CustomFiles Include="$(MSBuildProjectDirectory)/../images/**/*" />
  <DotNetPublishFiles Include="@(_CustomFiles)">
    <DestinationRelativePath>wwwroot/images/%(RecursiveDir)%(Filename)%(Extension)</DestinationRelativePath>
  </DotNetPublishFiles>
</ItemGroup>
```

The preceding markup:

* Can be added to the *.csproj* file or the publish profile. If it's added to the *.csproj* file, it's included in each publish profile in the project.
* Declares a `<_CustomFiles>` item to store files matching the `Include` attribute's globbing pattern. The *images* folder referenced in the pattern is located outside of the project directory. A [reserved property](/visualstudio/msbuild/msbuild-reserved-and-well-known-properties), named `$(MSBuildProjectDirectory)`, resolves to the project file's absolute path.

> * Uses the `DotNetPublishFiles` item, which is provided by a publish targets file in the Web SDK. The item accepts a list of files. The item's definition contains:

Reviewer: ```suggestion
* Uses the `<DotNetPublishFiles>` item, which is provided by a publish targets file in the Web SDK. The item accepts a list of files. The item's definition contains:
```

Reviewer 2: I'd actually go in the opposite direction here and never use brackets when describing an item by name.

>   * An `<DestinationRelativePath>` element whose inner text is empty. The inner text is overridden in the markup and uses [well-known item metadata](/visualstudio/msbuild/msbuild-well-known-item-metadata) such as `%(RecursiveDir)`. The inner text represents the *wwwroot/images* folder of the publish site.

Reviewer: ```suggestion
 * A `<DestinationRelativePath>` element whose inner text is empty. The inner text is overridden in the markup and uses [well-known item metadata](/visualstudio/msbuild/msbuild-well-known-item-metadata), such as `%(RecursiveDir)`. The inner text represents the *wwwroot/images* folder of the published site.
```

Reviewer 2: In general, I would prefer not mentioning the definition explicitly, instead referring to the "default behavior" of the item. Here, I don't think it's interesting that the default is empty, but it _is_ interesting to have an example of specifying the `DestinationRelativePath`.

  * An `<Exclude>` element whose inner text is `False`. No override is provided in the markup because file inclusion is the intent.

> ### Selective file inclusion

Author: I'm open to suggestions for this heading.

Reviewer: I like the heading ... I understand what's in the section.

The following example's `<ItemGroup>` element demonstrates:

> * Copying a file located outside of the project into the publish site's *wwwroot* folder. The file name of *ReadMe2.md* is maintained.

Reviewer: ```suggestion
* Copying a file located outside of the project into the published site's *wwwroot* folder. The file name of *ReadMe2.md* is maintained.
```

    <DeleteExistingFiles>False</DeleteExistingFiles>
  </PropertyGroup>
  <ItemGroup>
    <ResolvedFileToPublish Include="..\ReadMe2.md">
      <RelativePath>wwwroot\ReadMe2.md</RelativePath>
    </ResolvedFileToPublish>

    <Content Update="wwwroot\Content\**\*" CopyToPublishDirectory="Never" />
    <Content Update="Views\Home\About2.cshtml" CopyToPublishDirectory="Never" />
  </ItemGroup>
</Project>
```

> The preceding example uses the `ResolvedFileToPublish` item, which is provided by a publish targets file in the .NET Core SDK. The item's definition contains a `<CopyToPublishDirectory>` element whose inner text is `Always`. No `<CopyToPublishDirectory>` override is provided in the markup because file inclusion in the publish directory is the intent.

Reviewer: ```suggestion
The preceding example uses the `<ResolvedFileToPublish>` item, which is provided by a publish targets file in the .NET Core SDK. The item's definition contains a `<CopyToPublishDirectory>` element whose inner text is `Always`. No `<CopyToPublishDirectory>` override is provided in the markup because file inclusion in the publish directory is the intent.
```
